### PR TITLE
Ft google auth #187419170

### PR DESCRIPTION
**Pull request by MugemaneBertin2001**
# what does this PR do?
- Help user to sign up and login with existing google account
# Description of Task to be completed:
- As a user I should be able to athenticate (register/login) via Google
# how to test it 
- go to ft-google-auth-#187419170 branch
- clone this repo 
- create and fill .env with local configuration
- run the app 
- go to browser : localhost:3000/login
- click on URL for login with google
- follow on screen instruction

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
\ No newline at end of file
+JWT_SECRET = ""-->TODO: put your own jsonwebtoken scret  here
+
+GOOGLE_CLIENT_ID="" use yours
+GOOGLE_CLIENT_SECRET="" use yours
+GOOGLE_CALLBACK_URL="" use yours
+
+CLOUDINARY_CLOUD_NAME= "" use yours
+CLOUDINARY_API_KEY= "" use yours
+CLOUDINARY_API_SECRET= "" use yours
```
Comment (teerenzo) [inline .env.example]: we don't need this

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+            "server": "localhost",
+            "port": 5432,
+            "driver": "PostgreSQL",
+            "name": "new",
+            "group": "new",
+            "database": "ecommerce_db",
+            "username": "postgres"
+        }
+    ]
+}
```
Comment (teerenzo) [inline .vscode/settings.json]: please pay attention

File: __test__/home.test.ts [hunk 3]
```diff
@@ -12,9 +12,15 @@ describe("Testing Home route", () => {
...
-  test("servr should return status code of 200 --> given'/'", async () => {
+  test("server should return status code of 200 --> given'/'", async () => {
     const response = await request(app).get("/");
 
+    expect(response.status).toBe(200);
+  }, 20000);
+  test("server should return status code of 200 and link to log in with google --> given'/login'", async () => {
+    const response = await request(app).get("/login");
+    expect(response.text).toBe("<a href='/auth/google'>click to here to Login</a>")
+
```
Comment (teerenzo) [inline __test__/home.test.ts]: you should test what you have done not what you use to test it

File: src/utils/server.ts [hunk 4]
```diff
@@ -3,16 +3,34 @@ import cors from "cors";
...
+  
+  app.use(passport.initialize());
+  app.use(passport.session());
+
 app.use(cors());
 
 app.use("/", homeRoute);
 app.use("/api/v1", appROutes);
 app.use("/docs", docRouter);
+app.use('/', googleRoutes)
```
Comment (teerenzo) [inline src/utils/server.ts]: remove this here,

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Comment by teerenzo** [comment]
PR name and description is wrong please check on naming convertion again

File: .env.example (hunk 1 above)
Comment (MugemaneBertin2001) [reply .env.example]: removed it

File: .vscode/settings.json (hunk 2 above)
Comment (MugemaneBertin2001) [reply .vscode/settings.json]: added .vscode in .gitignore and removed it

File: __test__/home.test.ts (hunk 3 above)
Comment (MugemaneBertin2001) [reply __test__/home.test.ts]: added tests for my endpoints

File: src/utils/server.ts (hunk 4 above)
Comment (MugemaneBertin2001) [reply src/utils/server.ts]: refined as requested

**Comment by teerenzo** [comment]
you have many commits bertin squash them and are you sure you have changed 40 files

**Pull request closed without merging**